src/openshift/component.ts

```diff
@@ -148,13 +149,27 @@ export class Component extends OpenShiftItem {
         );
     }
 
+    static getPushCmd(): Thenable< string | undefined> {
+        return contextGlobalState.globalState.get('PUSH');
+    }
+
+    static setPushCmd(component, application, project): Thenable<void> {
+        return contextGlobalState.globalState.update('PUSH', `odo push ${component} --app ${application} --project ${project}`);
+    }
+
     static async push(context: OpenShiftObject): Promise<string> {
-        const component = await Component.getOpenShiftCmdData(context,
-            "In which Project you want to push the changes",
-            "In which Application you want to push the changes",
-            "For which Component you want to push the changes");
-        if (!component) return null;
-        Component.odo.executeInTerminal(Command.pushComponent(component.getParent().getParent().getName(), component.getParent().getName(), component.getName()));
+        const getPushCmd = await Component.getPushCmd();
+        if (getPushCmd && !context) {
```

Comment:
With this check in place there is no way to use push command from command palette. After first push it will always push the same component until context menu for component in tree is used.
You need to create new command 'OpenShift: Repeat last push command'.